### PR TITLE
#159965318 Estimated time to read an article

### PR DESCRIPTION
**Pull request by pngg**
#### What does this PR do?
Adds an estimated time to read an article

#### Description of Task to be completed
- Add estimated time to read an article so that a user will be able to see the time it takes to read an article

#### How should this be tested manually
- After cloning this repository, `git checkout ft-article-read-time-159965318`
- Activate the virtual environment with `. env/bin/activate`
- Install project requirements with `pip install -r requirements.txt`

`Use Insomnia/ Postman`

- Create a new article on this endpoint: `http://localhost:8000/api/articles/`

```
payload

{
    "slug": "how-to-train-your-dragon",
    "title": "How to train your dragon",
    "description": "Ever wonder how?",
    "body": "It takes a Jacobian It takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt ",
    "tagList": ["dragons", "training"],
    "favorited": false,
    "favoritesCount": 0,
    "author": {
      "username": "jake",
      "bio": "I work at statefarm",
      "image": "https://i.stack.imgur.com/xHWG8.jpg",
      "following": false
    }
}

Result

{
	"Article": {
		"id": 12,
		"title": "How to train your dragon",
		"body": "It takes a Jacobian It takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt takes a JacobianIt",
		"description": "Ever wonder how?",
		"created_at": "2018-09-18T11:16:52.393000+03:00",
		"updated_at": "2018-09-18T11:16:52.393031+03:00",
		"published_at": "2018-09-18T11:16:52.393042+03:00",
		"slug": "how-to-train-your-dragon-e9f0dd",
		"image": "/media/static/images/no-img.jpg",
		"author": 1,
		"favourited": [],
		"favourite": false,
		"favouritesCount": 0,
		"userLikes": [],
		"userDisLikes": [],
		"rating_average": null,
		"time_to_read": "0 min"
	}
}
```

#### What are the relevant pivotal tracker stories?
[Users should be able to see the time it takes to read an article](https://www.pivotaltracker.com/story/show/159965318)

### CODE REVIEW

File: authors/apps/articles/models.py [hunk 1]
```diff
@@ -47,6 +47,16 @@ def save(self, *args, **kwargs):
                                 uuid.uuid4().hex[:6])
         super().save(*args, **kwargs)
 
+    @property
+    def get_time_to_read(self):
+        # Set the standard read time
+        words_per_min = 270
```
Comment (pabishai) [inline authors/apps/articles/models.py]: @pngg Where is the time estimate for words per minute derived from

**Review by pabishai — COMMENTED** [review]
@pngg A review to consider

**Comment by micahoriaso** [comment]
@pngg, the minimum time to read as implemented is currently zero minutes, this gives the user a false impression of the level of effort it takes to read an article. I suggest that you make this value one minute as it is more representative.

File: authors/apps/articles/models.py [hunk 2]
```diff
@@ -47,6 +47,16 @@ def save(self, *args, **kwargs):
...
+        # Cleaning the post content and lower casing all words
+        for char in '-.,\n':
```
Comment (micahoriaso) [inline authors/apps/articles/models.py]: @pngg please give us descriptive comments about what is going on in this piece of code

**Review by micahoriaso — CHANGES_REQUESTED** [review]

File: authors/apps/articles/models.py (hunk 1 above)
Comment (pngg) [reply authors/apps/articles/models.py]: @PAbishai some googling :) and a bit of this https://www.jstor.org/stable/1161646. I intend to move this setting to ENV for easy manipulation/change.

File: authors/apps/articles/models.py (hunk 2 above)
Comment (pngg) [reply authors/apps/articles/models.py]: On it

**Comment by pngg** [comment]
> @pngg, the minimum time to read as implemented is currently zero minutes, this gives the user a false impression of the level of effort it takes to read an article. I suggest that you make this value one minute as it is more representative.

I agree. I am thinking of implementing a condition to return `Less than a min` if time is less than 1.

**Review by micahoriaso — APPROVED** [review]

File: authors/apps/articles/models.py [hunk 3]
```diff
@@ -47,6 +49,26 @@ def save(self, *args, **kwargs):
...
         super().save(*args, **kwargs)
 
+    @property
+    def get_time_to_read(self):
+        # Set the standard read time
+        words_per_min = settings.WORDS_PER_MIN
+        """
+        Cleaning the post content and lower casing all words: 
+        """
+        for char in '-.,\n':
```
Comment (parseen254) [inline authors/apps/articles/models.py]: @pngg this is an expensive operation. Replace it with regex. Resource: https://stackoverflow.com/questions/875968/how-to-remove-symbols-from-a-string-with-python

**Review by parseen254 — CHANGES_REQUESTED** [review]

File: authors/apps/articles/models.py (hunk 3 above)
Comment (pngg) [reply authors/apps/articles/models.py]: I have implemented the recommendation 👍

**Review by parseen254 — APPROVED** [review]

**Pull request merged**